sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -2789,11 +2762,16 @@ public PDone expand(PCollection<TableRow> input) {
       tagged
           .setCoder(KvCoder.of(ShardedKeyCoder.of(StringUtf8Coder.of()), TableRowInfoCoder.of()))
           .apply(Reshuffle.<ShardedKey<String>, TableRowInfo>of())
-          .apply(ParDo.of(new StreamingWriteFn(
-              tableSchema,
-              createDisposition,
-              tableDescription,
-              bqServices)));
+          .apply(
+              ParDo.of(
+                  new StreamingWriteFn(
+                      write.getJsonSchema() == null
+                          ? null
+                          : NestedValueProvider.of(
+                              write.getJsonSchema(), new JsonSchemaToTableSchema()),
```

Comment:
Pull this conditional assignment into a member variable  - kinda hard to read here.